mobile/src/home-view.js

```diff
@@ -295,6 +303,84 @@ class HomeView extends PureComponent {
     )
   }
 
+  renderSearch = () => {
+    const newStyle = {
+      flex: 1,
+      fontSize: 18,
+      color: '#364247',
+      textAlignVertical: 'top',
+      maxHeight: 100,
+      height: Math.max(35, this.state.inputHeight),
+      paddingTop: 0
+    }
+    const androidStyle = {
+      paddingLeft: 0,
+      marginTop: 5,
+      marginBottom: 5,
+    }
+    const iosStyle = {
+      marginTop: 3,
+    }
+    return (
+      <View
+      style={{
+        flexDirection: 'row',
+        backgroundColor: '#FFFFFF',
+        borderBottomColor: '#b7b7b7',
+        borderBottomWidth: 1,
+        borderRadius: 5,
+        height: 40,
+      }}
+    >
+      {this.state.search ? (
+        <View style={{ width: 40 }} />
+      ) : (
+        <TouchableOpacity style={s.circleBoxMargin}>
+          <Text style={s.whiteText}>?</Text>
+        </TouchableOpacity>
+      )}
+      <TextInput
+        style={Platform.select({
+          ios: [newStyle, iosStyle],
+          android: [newStyle, androidStyle],
+        })}
+        placeholder={t('search')}
+        value={this.state.lead}
+        onChangeText={lead => this.updateList(lead)}
+        maxLength={25}
+        placeholderTextColor="#9B9B9B"
+      />
+      {this.state.search ? (
+        <TouchableOpacity style={s.circleBoxMargin} onPress={this.resetSearch}>
+          <Text style={s.whiteText}>X</Text>
+        </TouchableOpacity>
+      ) : null}
+    </View>
+    )
+  }
+
+    updateList = value => {
+      const queryText = value ? value.toLowerCase() : ""
+      if (queryText.length > 0) {
+        const queryResult = []
+        this.state.cards.forEach(lead => {
+          const name = `${lead.firstName} ${lead.lastName}`
+          if (name) {
+            if (name.toLowerCase().indexOf(queryText) !== -1) {
+              queryResult.push(lead)
+            }
+          }
+        })
+        this.setState({ search: true, newList: queryResult, lead: value })
```

Comment:
This seems to be storing more info in `state` that is really necessary.  It seems like the search text is the only real state.  The filtered list can be derived from that query text in `render` based on that.  If the search text is the only state, then the unnecessary `newList` and `search` (boolean) can't get out of sync with it.